symphony/lib/toolkit/class.field.php

```diff
@@ -1538,10 +1538,25 @@ public function buildDSRetrievalSQL($data, &$joins, &$where, $andOperation = fal
     }
 
     /**
+     * Determine if the requested $order is random.
+     *
+     * @since Symphony 2.7.0
+     * @param string $order
+     *  the sorting direction.
+     * @return boolean
+     *  true if the $order is either 'rand' or 'random'
+     */
+    public static function isRandomOrder($order)
```

Comment:
> static: because it does not uses $this

You use this method only in non-static methods and inside the same object, therefore it make no sense to add `static`. (Yes, all method calls must be changed.)

> Why would it need to be an instance method ?

You have to ask the question differently: Why static? A static variable or method is always an exception. Static methods are nothing more than namespaced global functions.
The `Field` class is not stateless, why should this method be outside them?

> public: to be able to unit test it, one day...

Also no need for this, because you test (e.g.) `Field::buildSortingSQL()` and there you must test all parameters - included `random` and `rand`.
(You should not test protected or private members directly, but you can do this with PHP's [`ReflectionClass`](http://php.net/manual/en/class.reflectionclass.php).)